### PR TITLE
Users/daniellemott/v2 recent template

### PR DESCRIPTION
**Pull request by daniellemott**
**Static Frontend that displays Adaptive Card**

Static frontend that renders Adaptive Cards in the dashboard. Will connect to backend and implement design revisions in future PRs. 

To view changes:
1. cd adaptivecards-templates\private-templates-service\client\src
2. **Change** redirectUri in Config.tsx (adaptivecards-templates\private-templates-service\client\src) to http://localhost:3000
3. npm install
4. npm start

### CODE REVIEW

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import ReactDOM from "react-dom";
+
+import { Provider } from "react-redux";
+import { createStore } from "redux";
+import { rootReducer } from "../../store/rootReducer";
+import AdaptiveCard from "./AdaptiveCard";
+
+it("Renders without crashing", () => {
```
Comment (rachel-slater) [inline private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.test.tsx]: Just FYI - thorough unit tests will check if the each function you've written achieves it's purpose. 
For example, in your case, you might test whether the 'GetCard()' function returns the expected data object. 

e2e tests will come in really handy for what you're doing here - you can automate all of the 'actions' - i.e. opening a browser, clicking on a button, checking if your rendered card HTML is on the page - which is probably the most accurate way of testing this.

But for now, I wouldn't change anything - I think it's a little too early to be worrying about robust unit tests.

**Comment by daniellemott** [comment]
> can the tabs be changed to spaces?

done.

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import * as React from "react";
+import * as AdaptiveCards from "adaptivecards";
+import { Card, Container, TemplateName } from "../AdaptiveCard/styled";
+
+function getCard() {
+	// Hard coded, will remove and connect to backend in future PR
```
Comment (rachel-slater) [inline private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx]: FYI for the future, when you're removing the hard-coded data, you'll want to create an interface, which defines the structure of a card. For example:

```
interface CardData {
	type: boolean;
	version: string;
	body: any;
	actions: any;
}
```

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import * as React from "react";
+import * as AdaptiveCards from "adaptivecards";
+import { Card, Container, TemplateName } from "../AdaptiveCard/styled";
+
+function getCard() {
```
Comment (rachel-slater) [inline private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx]: As you're using Typescript, if the function returns something, you need to specify the _type_ of that return.

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	// Hard coded, will remove and connect to backend in future PR
+	var card = {
```
Comment (rachel-slater) [inline private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx]: Just [FYI](https://stackoverflow.com/questions/35572895/what-is-the-difference-between-var-and-let-in-typescript).

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+				type: "Action.OpenUrl",
+				title: "GitHub",
+				url: "http://github.com/Microsoft/AdaptiveCards"
+			}
+		]
+	};
+	return card;
+}
+
+function GetRenderedCard() {
```
Comment (rachel-slater) [inline private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx]: I would split this function up into multiple smaller functions, for example:
- renderingSetup()
- parseCardTemplate() - you'd also need to handle an error case here - what if the parse function couldn't parse the data object you give it?
- renderAdaptiveCard()

File: private-templates-service/client/src/components/Gallery/Gallery.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import * as React from "react";
+import AdaptiveCard from "../AdaptiveCard";
+import { Container } from "./styled";
+
+class Gallery extends React.Component {
+	render() {
+		return (
+			<Container>
+				<AdaptiveCard />
```
Comment (rachel-slater) [inline private-templates-service/client/src/components/Gallery/Gallery.tsx]: This 3 x <.AdaptiveCard /.> is just a temp thing for now right?

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.test.tsx (hunk 1 above)
Comment (daniellemott) [reply private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.test.tsx]: Okay, that makes sense. I will add a ticket to our backlog to add more robust unit tests once we have a more base.

File: private-templates-service/client/src/components/Gallery/Gallery.tsx (hunk 6 above)
Comment (daniellemott) [reply private-templates-service/client/src/components/Gallery/Gallery.tsx]: Yes, I will "un-hardcode" it once I start grabbing adaptive card templates from the database (via the backend).

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx (hunk 3 above)
Comment (daniellemott) [reply private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx]: Done.

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx (hunk 2 above)
Comment (daniellemott) [reply private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx]: Do you have any suggestions on how to make an interface for the adaptive card if I don't know what's going to be in the card?

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx (hunk 5 above)
Comment (daniellemott) [reply private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx]: I wasn't too sure about what I was doing, but added some types. Also, for the function "renderAdaptiveCard()" on line 59, I didn't know how to create an HTMLElement so I left the return type as any. Would love some more feedback!

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx (hunk 2 above)
Comment (rachel-slater) [reply private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx]: Ohh, I thought this was the standard structure across all Cards! If the object doesn't have consistent keys, then it won't make sense to base it on an interface. So it's good as it is :)

File: private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx (hunk 5 above)
Comment (rachel-slater) [reply private-templates-service/client/src/components/AdaptiveCard/AdaptiveCard.tsx]: You abstracted those into separate functions perfectly!! 👍 Much cleaner. And using 'any' is fine. (I had a quick search for a more appropriate HTML element type and couldn't find one either.)

**Pull request merged**